### PR TITLE
Add google authentication with oauth2-me gem.

### PR DESCRIPTION
**Pull request by freevova**

### CODE REVIEW

**Comment by EnragedWildkin** [comment]
@freevova Please rebase

**Comment by freevova** [comment]
@EnragedWildkin  Please review.

File: app/models/user.rb [hunk 1]
```diff
@@ -28,6 +28,18 @@ class User < ActiveRecord::Base
   enum role: [:registered, :admin]
   has_many :contracts
 
+  def self.from_omniauth(auth)
+    where(provider: auth.provider, uid: auth.uid).first_or_initialize.tap do |user|
```
Comment (EnragedWildkin) [inline app/models/user.rb]: Why do you need `tap`?

**Comment by EnragedWildkin** [comment]
@freevova Please check my comment.

**Comment by freevova** [comment]
@EnragedWildkin Please review.

**Comment by EnragedWildkin** [comment]
@freevova Looks good

**Pull request merged**